app/src/main/java/org/eyeseetea/malariacare/services/SurveyService.java

```diff
@@ -146,21 +152,35 @@ protected void onHandleIntent(Intent intent) {
 
     private void reloadDashboard() {
         Log.i(TAG, "reloadDashboard");
-        List<Survey> surveys = Survey.getAllSurveys();
+        IProgramRepository programLocalDataSource = new ProgramLocalDataSource();
+        IMainExecutor mainExecutor = new UIThreadExecutor();
+        IAsyncExecutor asyncExecutor = new AsyncExecutor();
+        GetUserProgramUIDUseCase getUserProgramUIDUseCase = new GetUserProgramUIDUseCase(
```

Comment:
Execution of this use case realize a query to database in every reload method, can you refactor for execute one time in service entry point (constructor onCreate)